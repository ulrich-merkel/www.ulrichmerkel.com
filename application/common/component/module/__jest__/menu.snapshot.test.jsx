/* eslint-disable import/no-extraneous-dependencies, func-names */
// @see https://github.com/airbnb/enzyme/issues/426
// @see https://github.com/facebook/jest/issues/1353
import React from 'react';
import renderer from 'react-test-renderer';
import ModuleMenu from '../menu';

describe('common/component/module/menu', function () {
    const defaultProps = {
        componentType: 'ol',
        className: 'module-menu',
        content: {
            name: 'bar',
            list: [
                {
                    path: '/foo',
                    title: 'title',
                    label: 'label',
                    itemType: 'foo',
                    icon: 'icon',
                    itemPropA: 'bar',
                    metaLinkUrl: '/test'
                },
                {
                    path: '/foo',
                    title: 'title',
                    isLabelHidden: true
                },
                {
                    path: '/foo',
                    title: 'title'
                }
            ]
        }
    };

    it('should render correctly', function () {
        const tree = renderer.create(
            <ModuleMenu
                {...defaultProps}
            >
                Module menu children
            </ModuleMenu>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should return null if content image is empty', function () {
        const tree = renderer.create(
            <ModuleMenu
                {...defaultProps}
                content={{
                    list: null
                }}
            >
                Module menu children not rendered
            </ModuleMenu>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should render no itemType if unset', function () {
        const tree = renderer.create(
            <ModuleMenu
                {...defaultProps}
                itemType={null}
            >
                Module menu children
            </ModuleMenu>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
