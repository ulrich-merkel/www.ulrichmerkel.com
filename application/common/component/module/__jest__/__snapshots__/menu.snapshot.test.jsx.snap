// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/component/module/menu should render correctly 1`] = `
<ol
  className="m-menu m-menu--bar module-menu"
  itemScope={true}
  itemType="https://schema.org/ItemList"
  role="menu"
>
  <li
    className="m-menu__list-item"
    itemProp="itemListElement"
    itemScope={true}
    itemType="foo"
  >
    <a
      aria-current={null}
      className="c-link m-menu__item"
      href="/foo"
      itemProp="bar"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      title="title"
    >
      <i
        aria-hidden="true"
        className="c-font-icon--icon m-menu__icon"
      />
      <span
        className="m-menu__label"
        itemProp="name"
      >
        label
      </span>
    </a>
    <link
      href="/test"
      itemProp="url"
    />
  </li>
  <li
    className="m-menu__list-item"
    itemProp="itemListElement"
    itemScope={true}
    itemType="http://www.schema.org/SiteNavigationElement"
  >
    <a
      aria-current={null}
      className="c-link m-menu__item"
      href="/foo"
      itemProp="url"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      title="title"
    >
      <span
        className="is-visually-hidden m-menu__label"
        itemProp="name"
      >
        
      </span>
    </a>
  </li>
  <li
    className="m-menu__list-item"
    itemProp="itemListElement"
    itemScope={true}
    itemType="http://www.schema.org/SiteNavigationElement"
  >
    <a
      aria-current={null}
      className="c-link m-menu__item"
      href="/foo"
      itemProp="url"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      title="title"
    >
      <span
        className="m-menu__label"
        itemProp="name"
      >
        
      </span>
    </a>
  </li>
  Module menu children
</ol>
`;

exports[`common/component/module/menu should render no itemType if unset 1`] = `
<ol
  className="m-menu m-menu--bar module-menu"
  role="menu"
>
  <li
    className="m-menu__list-item"
    itemProp="itemListElement"
    itemScope={true}
    itemType="foo"
  >
    <a
      aria-current={null}
      className="c-link m-menu__item"
      href="/foo"
      itemProp="bar"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      title="title"
    >
      <i
        aria-hidden="true"
        className="c-font-icon--icon m-menu__icon"
      />
      <span
        className="m-menu__label"
        itemProp="name"
      >
        label
      </span>
    </a>
    <link
      href="/test"
      itemProp="url"
    />
  </li>
  <li
    className="m-menu__list-item"
    itemProp="itemListElement"
    itemScope={true}
    itemType="http://www.schema.org/SiteNavigationElement"
  >
    <a
      aria-current={null}
      className="c-link m-menu__item"
      href="/foo"
      itemProp="url"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      title="title"
    >
      <span
        className="is-visually-hidden m-menu__label"
        itemProp="name"
      >
        
      </span>
    </a>
  </li>
  <li
    className="m-menu__list-item"
    itemProp="itemListElement"
    itemScope={true}
    itemType="http://www.schema.org/SiteNavigationElement"
  >
    <a
      aria-current={null}
      className="c-link m-menu__item"
      href="/foo"
      itemProp="url"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      title="title"
    >
      <span
        className="m-menu__label"
        itemProp="name"
      >
        
      </span>
    </a>
  </li>
  Module menu children
</ol>
`;

exports[`common/component/module/menu should return null if content image is empty 1`] = `null`;
